File: Aveline.Api/Modules/Commerce/Models/OrderItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Aveline.Api.Common.MultiTenancy;
using Aveline.Api.Modules.Organizations.Models;

namespace Aveline.Api.Modules.Commerce.Models;

[Table("Order_Items")]
public class OrderItem : ITenantEntity
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid OrganizationId { get; set; }

    [Required]
    public Guid OrderId { get; set; }

    [Required]
    public Guid ItemId { get; set; } // Pointing to Inventory slice

    [Required]
    [MaxLength(200)]
    public string ItemName { get; set; } = string.Empty;

    [Required]
    public int Quantity { get; set; } = 1;

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal UnitPrice { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal WholesaleCost { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal TotalPrice { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation property
    // Navigation properties
    public Organization? Organization { get; set; }
    [ForeignKey(nameof(OrderId))]
    public Order Order { get; set; } = null!;
}

File: frontend/aveline_mobile/lib/features/commerce/domain/entities/order_item.dart
class OrderItem {
  const OrderItem({
    this.id,
    required this.itemId,
    required this.itemName,
    this.quantity = 1,
    required this.unitPrice,
    this.wholesaleCost = 0.0,
    double? totalPrice,
  }) : totalPrice = totalPrice ?? (unitPrice * quantity);

  final String? id;
  final String itemId;
  final String itemName;
  final int quantity;
  final double unitPrice;
  final double wholesaleCost;
  final double totalPrice;

  double get margin => totalPrice - (wholesaleCost * quantity);
  double get marginPercent =>
      totalPrice > 0 ? ((totalPrice - (wholesaleCost * quantity)) / totalPrice) * 100 : 0.0;

  OrderItem copyWith({
    String? id,
    String? itemId,
    String? itemName,
    int? quantity,
    double? unitPrice,
    double? wholesaleCost,
    double? totalPrice,
  }) {
    final nextQty = quantity ?? this.quantity;
    final nextUnitPrice = unitPrice ?? this.unitPrice;
    return OrderItem(
      id: id ?? this.id,
      itemId: itemId ?? this.itemId,
      itemName: itemName ?? this.itemName,
      quantity: nextQty,
      unitPrice: nextUnitPrice,
      wholesaleCost: wholesaleCost ?? this.wholesaleCost,
      totalPrice: totalPrice ?? (nextUnitPrice * nextQty),
    );
  }
}
